CoreUtilities/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenu.java

```diff
@@ -84,6 +85,30 @@ private JDropDownMenu(final String text, final Icon icon, final List<E> items) {
             this.button = new JButton(text, arrow);
         }
         button.setHorizontalTextPosition(SwingConstants.LEFT);
+        addButtonActionListener(button, menu);
+        add(button);
+
+        listeners = new HashSet<>();
+
+        items.forEach(item -> {
+            final JMenuItem menuItem = new JMenuItem(item.toString());
+            menuItem.addActionListener(new OpenAction(menu, button));
+            addMenuItemActionListener(menuItem);
+            menuItems.put(menuItem, item);
+            menu.add(menuItem);
+        });
+    }
+
+    protected final void addMenuItemActionListener(final JMenuItem menuItem) {
+        menuItem.addActionListener(event -> {
+            final JMenuItem changedMenuItem = (JMenuItem) event.getSource();
+            final E changedItem = menuItems.get(changedMenuItem);
+            final ActionEvent actionEvent = new ActionEvent(changedItem, ActionEvent.ACTION_PERFORMED, null);
+            actionPerformed(actionEvent);
```

Comment:
a bit confused by this change. Previously this was done in a thread for each listener but that seems to have been omitted here so this would behave differently.

Reply:
The actionPerformed it calls executes in a thread for each listener:

> 
> 
>     public void actionPerformed(final ActionEvent event) {
>         listeners.forEach(listener -> {
>             final Thread thread = new Thread("JDropDownMenu Worker") {
>                 @Override
>                 public void run() {
>                     listener.actionPerformed(event);
>                 }
>             };
>             thread.start();
>         });
>     }

Reply:
gotcha. Didn't see that but that makes sense